cmd/av/stack_next.go

```diff
@@ -50,16 +44,29 @@ var stackNextCmd = &cobra.Command{
 			return err
 		}
 
-		// confirm we can in fact do the operation given current branch state
-		if len(subsequentBranches) == 0 {
-			return errors.New("there is no next branch")
-		}
-		if n > len(subsequentBranches) {
-			return fmt.Errorf("invalid number (there are only %d subsequent branches in the stack)", len(subsequentBranches))
+		var branchToCheckout string
+		if !stackNextFlags.Last {
+			if len(subsequentBranches) == 0 {
+				return errors.New("there is no next branch")
+			}
+			n, err := strconv.Atoi(args[0])
```

Comment:
You need to check the length of args here. Can probably just skip the `len(args)` check at the top and inline it here (and make sure `len(args)` is zero if `--last` is given).

So something like (in atrocious pseudocode):

```
if --last {
  assert len(args) == 0
  targetBranch = ...
} else {
  n = 1
  if len(args) == 1 {
    n = parseInt(args[0])
  } else if len(args) > 1 {
    return error
  }
}
```